### PR TITLE
CMM-927 support attachments limit

### PR DESCRIPTION
**Pull request by adalpari**
## Description
This PR checks the attachments size before sending them to the server, and shows proper errors if the user reaches the limit.

## Testing instructions

1. Log in with a WP.COM account and enable `MODERN_SUPPORT` FF
2. Create a new ticket or add messages to an existing one
- [ ] Check you can attach files, but if you hit the limit (20MB) you see a proper error

Screenshot_20251105_113238

Screenshot_20251105_115034

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>strings.xml</code> files should only be updated on release branches, when the translations are downloaded by our automation.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by adalpari** [comment]
@claude

File: WordPress/src/main/java/org/wordpress/android/support/he/ui/HESupportViewModel.kt [hunk 1]
```diff
@@ -159,15 +176,107 @@ class HESupportViewModel @Inject constructor(
...
+            val fileSize = getFileSize(uri)
+
+            // Skip if we can't determine file size
+            if (fileSize == null) {
+                skippedUris.add(uri)
+                continue
+            }
+
+            // Check individual file size
+            if (fileSize > MAX_TOTAL_SIZE_BYTES) {
```
[the remark this answers is not in the document]
Comment (adalpari) [reply WordPress/src/main/java/org/wordpress/android/support/he/ui/HESupportViewModel.kt]: This is a threshold the server is imposing, so we have no choice but to check the file size at the client side.

File: WordPress/src/main/java/org/wordpress/android/support/he/ui/HESupportViewModel.kt [hunk 2]
```diff
@@ -159,15 +176,107 @@ class HESupportViewModel @Inject constructor(
...
+                AttachmentState(
+                    acceptedUris = newAccepted,
+                    rejectedUris = emptyList(),
+                    rejectionReason = null
+                )
+            }
+        }
+    }
+
+    @Suppress("TooGenericExceptionCaught", "SwallowedException")
```
[the remark this answers is not in the document]
Comment (adalpari) [reply WordPress/src/main/java/org/wordpress/android/support/he/ui/HESupportViewModel.kt]: We are aware of that. I'm adding a log for internal purposes.

File: WordPress/src/main/java/org/wordpress/android/support/he/ui/HESupportViewModel.kt [hunk 3]
```diff
@@ -159,15 +176,107 @@ class HESupportViewModel @Inject constructor(
...
+            val fileSize = getFileSize(uri) ?: 0L
+            currentTotalSize += fileSize
+        }
+
+        // Validate each new attachment
+        for (uri in uris) {
+            val fileSize = getFileSize(uri)
+
+            // Skip if we can't determine file size
+            if (fileSize == null) {
```
[the remark this answers is not in the document]
Comment (adalpari) [reply WordPress/src/main/java/org/wordpress/android/support/he/ui/HESupportViewModel.kt]: This was a mistake, I'm now allowing them to be added and we will rely on the server check.

File: WordPress/src/main/java/org/wordpress/android/support/he/ui/HESupportViewModel.kt [hunk 4]
```diff
@@ -159,15 +176,107 @@ class HESupportViewModel @Inject constructor(
...
+            validUris.add(uri)
+            currentTotalSize += fileSize
+        }
+
+        // Build the new attachment state
+        val currentAccepted = _attachmentState.value.acceptedUris
+        val newAccepted = currentAccepted + validUris
+
+        when {
+            skippedDueToFileSize -> {
```
[the remark this answers is not in the document]
Comment (adalpari) [reply WordPress/src/main/java/org/wordpress/android/support/he/ui/HESupportViewModel.kt]: When a mixed case happens, the `skippedDueToTotalSize` has higher priority, so the user can focus on one error. After that, if they try to upload a file greater than 20MB then they can work on that specific file.

**Comment by adalpari** [comment]
> Issues Found:
> 
> 🔴 Critical: Files with unknown size are silently rejected without user feedback (HESupportViewModel.kt:202-206)
> 🟡 High: Only one rejection reason is shown when multiple rejection types occur (HESupportViewModel.kt:231-253)
> 🟡 Medium: Individual file size check may be too strict - rejects files ≥20MB instead of >20MB (HESupportViewModel.kt:209)
> 🟡 Medium: Exception swallowing without logging could mask issues (HESupportViewModel.kt:256-267)
> 🔵 Low: Missing test coverage for edge cases (null file size, exactly 20MB)
> 🔵 Enhancement: Consider showing count of skipped files in UI

Major issues fixed ([see the second Claude iteration](https://github.com/wordpress-mobile/WordPress-Android/pull/22338#issuecomment-3490637743))
There's a test issue I'm fixing now.

**Comment by adalpari** [comment]
I've changed a bit the flow to include the [progressbar iOS used](https://github.com/wordpress-mobile/WordPress-iOS/pull/24972). And now, every time the user removes a selected file, we try to re-add the skipped ones because of the total size.

https://github.com/user-attachments/assets/6b7add4e-62c2-4f82-9eea-62843ad92dd2

**Review by jkmassel — APPROVED** [review]
I tested this and it worked as described.
